modules/ip6/control/address.c

```diff
@@ -76,24 +76,25 @@ struct nexthop *ip6_mcast_get_member(uint16_t iface_id, const struct rte_ipv6_ad
 	return NULL;
 }
 
-static int ip6_mcast_addr_add(struct iface *iface, const struct rte_ipv6_addr *ip) {
+static int ip6_mcast_addr_add(struct iface *iface, const struct rte_ipv6_addr *ipv6) {
 	struct hoplist *maddrs = &iface_mcast_addrs[iface->id];
 	struct nexthop *nh = NULL;
+	struct rte_ipv6_addr ip = *ipv6;
 	unsigned i;
 
-	LOG(INFO, "%s: joining multicast group " IP6_F, iface->name, ip);
+	LOG(INFO, "%s: joining multicast group " IP6_F, iface->name, &ip);
 
 	for (i = 0; i < maddrs->count; i++)
-		if (rte_ipv6_addr_eq(&maddrs->nh[i]->ipv6, ip))
+		if (rte_ipv6_addr_eq(&maddrs->nh[i]->ipv6, &ip))
 			return errno_set(EEXIST);
 
 	if (i == ARRAY_DIM(maddrs->nh))
 		return errno_set(ENOSPC);
 
-	if ((nh = ip6_nexthop_lookup(iface->vrf_id, ip)) == NULL) {
-		if ((nh = ip6_nexthop_new(iface->vrf_id, GR_IFACE_ID_UNDEF, ip)) == NULL)
+	if ((nh = ip6_nexthop_lookup(iface->vrf_id, &ip)) == NULL) {
+		if ((nh = ip6_nexthop_new(iface->vrf_id, GR_IFACE_ID_UNDEF, &ip)) == NULL)
 			return errno_set(-errno);
-		rte_ether_mcast_from_ipv6(&nh->lladdr, ip);
+		rte_ether_mcast_from_ipv6(&nh->lladdr, &ip);
```

Comment:
I think I missed something. Why are you introducing a new local ip variable to use it instead of the const parameter?